back/app/controllers/web_api/v1/internal_comments_controller.rb

```diff
@@ -103,6 +103,16 @@ def update
     end
   end
 
+  def mark_as_deleted
```

Comment:
Actually, the RESTful way of implementing "mark as deleted" would be to use PATCH to set the `publication_status` to 'deleted'.